wrappers/csharp/src/Managed.cs

```diff
@@ -634,6 +645,94 @@ public static byte[] Encrypt(byte[] data, byte[] key, CipherTextVersion version
             return result;
         }
 
+        /// <summary>
+        /// Sign data using a keypair to certify its authenticity.
+        /// </summary>
+        /// <param name="data">The data to sign.</param>
+        /// <param name="keypair">The keypair to use to sign the data.</param>
+        /// <param name="version">The signature version to use. (Latest is recommended).</param>
+        /// <returns>Returns the signature result as byte array.</returns>
+        public static byte[] Sign(byte[] data, SigningKeyPair keypair, SignatureVersion version = SIGNATURE_VERSION)
+        {
+            if (data == null || data.Length == 0)
+            {
+                return null;
+            }
+
+            if (keypair == null)
+            {
+                throw new DevolutionsCryptoException(ManagedError.InvalidParameter);
+            }
+
+            byte[] keypairNative = keypair.ToByteArray();
+
+            byte[] result = new byte[Native.SignSize((ushort)version)];
+
+            long resultLength = Native.Sign(data, (UIntPtr)data.Length, keypairNative, (UIntPtr)keypairNative.Length, result, (UIntPtr)result.Length, (ushort)version);
+
+            if (resultLength < 0)
+            {
+                Utils.HandleError(resultLength);
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Verify some data using a signature and the corresponding public key.
+        /// </summary>
+        /// <param name="data">The data to verify.</param>
+        /// <param name="publicKey">The public key that was used to sign the data.</param>
+        /// <param name="signature">The signature to verify.</param>
+        /// <returns>Returns false if the data, the signature or the public key is invalid or true if everything is valid.</returns>
+        public static bool VerifySignature(byte[] data, SigningPublicKey publicKey, byte[] signature)
+        {
+            if (data == null || data.Length == 0)
+            {
+                return false;
+            }
+
+            if (publicKey == null || signature == null || signature.Length == 0)
+            {
+                throw new DevolutionsCryptoException(ManagedError.InvalidParameter);
+            }
+
+            byte[] publicKeyNative = publicKey.ToByteArray();
+
+            long res = Native.VerifySignature(data, (UIntPtr)data.Length, publicKeyNative, (UIntPtr)publicKeyNative.Length, signature, (UIntPtr)signature.Length);
+
+            if (res == 0)
+            {
+                return false;
+            }
+
+            if (res < 0)
+            {
+                Utils.HandleError(res);
+            }
+
+            return true;
```

Comment:
I'd rather have an explicit `if (res == 1) {return true}` then a "fallback" like this to avoid false positives. We really don't want to "accept" an invalid signature.